x-pack/plugins/observability_solution/apm_data_access/server/plugin.ts

```diff
@@ -32,16 +34,32 @@ export class ApmDataAccessPlugin
     const apmDataAccessConfig = this.initContext.config.get<APMDataAccessConfig>();
     const apmIndicesFromConfigFile = apmDataAccessConfig.indices;
 
+    const getApmIndices = async (savedObjectsClient: SavedObjectsClientContract) => {
+      const apmIndicesFromSavedObject = await getApmIndicesSavedObject(savedObjectsClient);
+      return { ...apmIndicesFromConfigFile, ...apmIndicesFromSavedObject };
+    };
+
     // register saved object
     core.savedObjects.registerType(apmIndicesSavedObjectDefinition);
 
     // expose
+    const getCoreStart = () => core.getStartServices().then(([coreStart]) => coreStart);
+    const getResourcesForServices = async () => {
+      const coreStart = await getCoreStart();
+      const soClient = coreStart.savedObjects.createInternalRepository();
```

Comment:
Have you tried changing the APM index patterns on the APM settings page and see if the correct indices are returned with this `createInternalRepository`?

Reply:
Just tested it and it worked. APM also uses it here [here](https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability_solution/apm/server/plugin.ts#L120). It could be due to the fact that when APM indices are changed on the Settings page, Kibana does a full page refresh.